source/core_json.c

```diff
@@ -335,6 +335,9 @@ static bool skipOneHexEscape( const char * buf,
 
     i = *start;
 #define HEX_ESCAPE_LENGTH    ( 6U )   /* e.g., \u1234 */
+    /* MISRA Ref 14.3.1 [Configuration dependent invariant] */
+    /* More details at: https://github.com/FreeRTOS/coreJSON/blob/main/MISRA.md#rule-143 */
+    /* coverity[misra_c_2012_rule_14_3_violation] */
```

Comment:
I am not sure I understand what is invariant here. `SIZE_MAX - HEX_ESCAPE_LENGHT` is a constant.
And `i` is size_t which means that it can go beyond that constant. I do no think that it is invariant.
Am I missing something?

Reply:
Can we refactor the function such that the SIZE_MAX overflow check is never required. The `max` variable already holds the bound up to which the parser can move. Could we simplify bounds checking using max variable as follows?

`
if ( ( max >= HEX_ESCAPE_LENGTH )  && (  i <= ( max - HEX_ESCAPE_LENGTH ) )  )
{
    end =  ( i + HEX_ESCAPE_LENGTH );
}
else
{
    end  = max;
}
`
With this we can probably remove `end < max` check in line 343.